mmit/core/piecewise_function.cpp

```diff
@@ -165,7 +165,7 @@ int PiecewiseFunction::insert_point(double y, bool is_upper_bound) {
     }
 
     // Breakpoint info
-    float b = y - s * this->margin;
```

Comment:
wow, sorry I didn't catch that -- in general I never use float -- that should have raised a red flag in my head -- I completely agree, change to using double all the time.